[web] Added guide for testing web performance

EvanBacon opened the pull request.

EvanBacon committed:
Added guide for testing web performance

ide commented on docs/pages/versions/unversioned/guides/web-performance.md:
```diff
@@ -0,0 +1,73 @@
+Lighthouse is a great way to see how fast, accessible, and performant your website is.
+You can test your project with the _Audit_ tab in Chrome, or with the [**Lighthouse CLI**][lighthouse].
+
+After creating a production build with `expo build:web` and serving somewhere. Run lighthouse with the URL your site is hosted at.
```
```suggestion
After creating a production build with `expo build:web` and serving it somewhere, run Lighthouse with the URL your site is hosted at.
```

EvanBacon committed:
[storybook] deleted Loki (#4372)

* Delete .gitignore

* Delete chrome_laptop_APIs_Constants.png

* Delete chrome_laptop_Components_BlurView.png

* Delete chrome_laptop_Components_Linear_Gradient.png

* Update config.js

* Update package.json

EvanBacon committed:
[storybook] Fix component packages (#4373)

* Update package.json

* Update yarn.lock

* Update BlurViewScreen.js

* Update LinearGradientScreen.js

* Update BlurViewScreen.js

* Update LinearGradientScreen.js

* Update LinearGradientScreen.js

tsapeta committed:
Automatically add 'bug' label to bug reports

EvanBacon committed:
[storybook] Jest snapshots (#4379)

* added jest snapshots

* Removed loki command

* deleted loki

* Update babel.config.js

* Update config.js

* Update addons.js

* Update babel.config.js

* [WIP]

* Fixed storybook in workspace

* Update config.js

* Update package.json

* Update package.json

* Update package.json

* Update yarn.lock

* Update storyshots.test.js

* Create storyshots-test-js-image-ap-is-constants-1-snap.png

* Create storyshots-test-js-image-components-blur-view-1-snap.png

* Create storyshots-test-js-image-components-linear-gradient-1-snap.png

* Update storyshots.test.js.snap

* Added storybook test to CI

* Update config.yml

* Update storyshots-test-js-image-components-blur-view-1-snap.png

* Update storyshots.test.js

* Update package.json

* Added image params for CI

* Update storyshots.test.js.snap

* Update storyshots.test.js

* Update black-128x256.png

* Update example_image_1.jpg

* Update exponent-icon@2x.png

* Update exponent-icon@3x.png

* Update app.png

* Update loading.png

* Update qrcode_photo_wikipedia.jpg

* Update roadster.jpg

* Update package.json

* Update BlurViewScreen.js

* Update LinearGradientScreen.js

* Update package.json

* Update storyshots-test-js-image-components-blur-view-1-snap.png

* Update BlurViewScreen.js

* Update LinearGradientScreen.js

* Update AppText.js

* Update Code.js

* Update DocItem.js

* Create rem.js

* Update Section.js

* Update StyleList.js

* Update TextList.js

* Update UIExplorer.js

* Update package.json

* Update storyshots.test.js.snap

* Made style more generic

* Disabled visual regression testing in CI

tsapeta committed:
[ci] add shell_app_ios_build job to expo repo (#4355)

# Why

Shell apps could be built on the CI of this repo instead of the universe. 

# How

I've moved building iOS shell apps from the universe and also merged two jobs (for simulator and for real devices) into one job `shell_app_ios_build` that also uploads the tarball at the end.
I also placed iOS and Android jobs under one workflow `shell_app` that will be available only on the release branches and branches with `all-ci` substring (for testing) - I don't think we need to build shell apps on every commit as it happened so far (for iOS simulator), but I'm open to discussion 😉 

_bonus:_ Also fixed a bug in `expo_client_build` job that caused the tarball not to be uploaded to S3 😅 

# Test Plan

I've been testing this new `shell_app` workflow on my `@tsapeta/all-ci` branch, where I successfully built iOS shell app.

tsapeta committed:
[expotools] add update-turtle-sdk-version command (#4381)

# Why

Moving things from universe to the open repo.

# How

Moved `update-turtle-sdk-version` script from powertools in universe to expotools.

# Test Plan

Successfully updated Turtle SDK version to SDK33.

satuevsky committed:
[android][expo-av] Fix `AudioFocusNotAcquiredException` when playing from background (#4361)

If `mStaysActiveInBackground` is `true` then don't throw the `AudioFocusNotAcquiredException`.

tsapeta committed:
[module-scripts] update jest-preset to include transforms from jest-expo

tsapeta committed:
[expo-camera] fix 'yarn build' returning TypeScript errors

sjchmiela committed:
[ci] Add FASTLANE_SKIP_UPDATE_CHECK env var to CI (#4384)

Up until now a non-latest version of Fastlane made the job fail (return code was 0).

tsapeta committed:
[contacts] fix 'yarn build' returning TypeScript errors

tsapeta committed:
[ios] fix react-native-screens on SDK32

Fixes #4371

tsapeta committed:
[changelog] cut off 33.0.0 version in changelog

esamelson committed:
[README] add yarn build step to readme

EvanBacon committed:
[storybook] Migrate components (#4391)

* Update yarn.lock

* Update jest.config.web.js

* Update UIExplorer.js

* Create Table.js

* Create ListButton.js

* Update index.js

* Update DocItem.js

* Added storybook components

* [WIP] update casing names

* Revert category names

* Update storyshots.test.js.snap

cruzach committed:
Docs: clarify supported node versions  (#4352)

* docs: clarify supported node versions

* integrate James's Comments

* backport SDK32

* backport SDK31

SimenB committed:
`registerRootComponent` should accept FunctionalComponent (#4387)

# Why

Allow passing `FunctionalComponent` to `registerRootComponent`, not just `ClassComponent`

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/21e68259f6fd43aba37e8112e3edb792975f8266/types/react/index.d.ts#L73

# How

Edited on GitHub

# Test Plan

No type error passing e.g. `const App: React.FC = () => <View />;`

EvanBacon committed:
Create random.md (#4398)

EvanBacon committed:
Create crypto.md (#4397)

EvanBacon committed:
Update docs/pages/versions/unversioned/guides/web-performance.md

Co-Authored-By: James Ide <ide@users.noreply.github.com>

EvanBacon committed:
Merge branch 'master' into @evanbacon/expo-web/docs/performance

ide reviewed (approved):
Not sure what happened to your commit history but you might want to rebase and squash from the command line manually to clean it up.
